scripts/run_backend_tests_test.py

```diff
@@ -0,0 +1,888 @@
+            'Failed to calculate coverage because subprocess failed. '
+            '%s' % failed_process_output)
+        with swap_subprocess_run, self.assertRaisesRegex(
+                RuntimeError, error_msg):
+            run_backend_tests._check_coverage(True) # type: ignore[no-untyped-call] # pylint: disable=protected-access
+
+    def test_coverage_is_calculated_correctly_for_specific_files(self) -> None:
+        with self.swap_install_third_party_libs:
+            from scripts import run_backend_tests
+        include_files = (
+            'scripts/run_backend_tests.py', 'core/domain/exp_domain.py')
+        self.coverage_check_cmd.append('--include=%s' % ','.join(include_files))
+        coverage_report_output = 'TOTAL       283     36    112     10    86% '
+        process = MockProcessOutput()
+        process.stdout = coverage_report_output
+        def mock_subprocess_run(
+                cmd: Any, **unused_kwargs: Any) -> MockProcessOutput:
+            if cmd == self.coverage_combine_cmd:
+                return MockProcessOutput()
+            elif cmd == self.coverage_check_cmd:
+                return process
+            else:
+                raise Exception(
+                    'Invalid command passed to subprocess.run() method')
+
+        swap_subprocess_run = self.swap(subprocess, 'run', mock_subprocess_run)
+        with swap_subprocess_run:
+            returned_output, coverage = run_backend_tests._check_coverage( # type: ignore[no-untyped-call] # pylint: disable=protected-access
+                True, include=include_files)
+
+        self.assertEqual(returned_output, coverage_report_output)
+        self.assertEqual(coverage, 86)
+
+    def test_coverage_is_calculated_correctly_for_a_single_file(self) -> None:
+        with self.swap_install_third_party_libs:
+            from scripts import run_backend_tests
+        data_file = '.coverage.hostname.12345.987654321'
+        coverage_report_output = 'TOTAL       283     36    112     10    86% '
+        process = MockProcessOutput()
+        process.stdout = coverage_report_output
+        def mock_subprocess_run(
+                cmd: Any, **unused_kwargs: Any) -> MockProcessOutput:
+            if cmd == self.coverage_combine_cmd:
+                return MockProcessOutput()
+            elif cmd == self.coverage_check_cmd:
+                return process
+            else:
+                raise Exception(
+                    'Invalid command passed to subprocess.run() method')
+
+        swap_subprocess_run = self.swap(subprocess, 'run', mock_subprocess_run)
+        with swap_subprocess_run:
+            returned_output, coverage = run_backend_tests._check_coverage( # type: ignore[no-untyped-call] # pylint: disable=protected-access
+                False, data_file=data_file)
+
+        self.assertEqual(returned_output, coverage_report_output)
+        self.assertEqual(coverage, 86)
+
+    def test_no_data_to_report_returns_full_coverage(self) -> None:
```

Comment:
This behavior seems wrong. Shouldn't we return a coverage of 0 if there is no data to report? If you agree, we can open an issue--no need to fix it in this PR

Reply:
This is useful when the file is passed with an `--omit` flag. [Context](https://github.com/oppia/oppia/blob/develop/scripts/run_backend_tests.py#L582)

Reply:
OK, that sounds good